### PR TITLE
improved error message.

### PR DESCRIPTION
**Pull request by k1nshuk**
## Fixes/Resolves:

Provides better instructions for debugging scaling problems, e.g., 0/0 calculations to identify which function the error is coming from.

## Summary/Motivation:
Making the debugging process easier

## Changes proposed in this PR:
- Better error message. See file for changes

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by bknueven — APPROVED** [review]

File: watertap/core/plugins/solvers.py [hunk 1]
```diff
@@ -158,7 +158,9 @@ def _scale_constraints(self, blk):
                     self._cleanup()
                     raise RuntimeError(
                         "Error in AMPL evaluation.\n"
-                        "Run ipopt with halt_on_ampl_error=yes and symbolic_solver_labels=True to see the affected function."
+                        "Re-run ipopt with:\n"
+                        '1. solver options = {"halt_on_ampl_error" : "yes", "nlp_scaling_method" : "gradient-based"\n'
```
Comment (adam-a-a) [inline watertap/core/plugins/solvers.py:162]: I am forgetting- should `nlp_scaling_method` be what you have here or `user-scaling`? What happens if you don't specify gradient -based scaling?

Comment (bknueven) [reply watertap/core/plugins/solvers.py:162]: This option ensures watertap's own scaling gets skipped and the model gets handed directly to Ipopt. This is what we want to see the problematic constraint / expression.

Comment (adam-a-a) [reply watertap/core/plugins/solvers.py:162]: I just thought I remembered this working regardless of that option so maybe I'm not recalling correctly.

Comment (bknueven) [reply watertap/core/plugins/solvers.py:162]: It certainly does in IDAES or if you just use `SolverFactory("ipopt")`.

**Review by adam-a-a — APPROVED** [review]

**Pull request merged**